### PR TITLE
update aws-sdk gem to v3

### PR DESCRIPTION
**Pull request by AlexanderZagaynov**
This is a part of global update `aws-sdk` gem to `v3`

Together with:
- https://github.com/ManageIQ/manageiq-providers-amazon/pull/488
- https://github.com/ManageIQ/amazon_ssa_support/pull/39

It should fix https://github.com/ManageIQ/manageiq-providers-amazon/issues/465

### CODE REVIEW

File: manageiq-gems-pending.gemspec [hunk 1]
```diff
@@ -24,7 +24,7 @@ Gem::Specification.new do |s|
   s.add_runtime_dependency "activerecord",            ">= 5.0", "< 5.2"
   s.add_runtime_dependency "activesupport",           ">= 5.0", "< 5.2"
   s.add_runtime_dependency "awesome_spawn",           "~> 1.4"
-  s.add_runtime_dependency "aws-sdk",                 "~> 2.9.7"
+  s.add_runtime_dependency "aws-sdk",                 "~> 3"
```
Comment (Fryguy) [inline manageiq-gems-pending.gemspec]: Please spell out the full gem version, otherwise we run the risk of accidentally pulling in v4 when it's released.

Comment (AlexanderZagaynov) [reply manageiq-gems-pending.gemspec]: done

**Comment by carbonin** [comment]
cc @NickLaMuro regarding https://github.com/ManageIQ/manageiq-gems-pending/pull/361/files#r216466360

**Comment by NickLaMuro** [comment]
Oh... I didn't realize this was part of `manageiq-gems-pending`.  Well that makes things easier.

@AlexanderZagaynov can you delete `lib/gems/pending/util/extensions/aws-sdk/s3_upload_stream_patch.rb` as part of this patch?  I think there is one other file in `manageiq-providers-amazon` that I have also patch that should be in the newest version of the gem (right @Fryguy ?  it was your patch I believe).

**Comment by NickLaMuro** [comment]
And just an FYI...

https://github.com/ManageIQ/manageiq-gems-pending/pull/361/files#r219526521

**Comment by AlexanderZagaynov** [comment]
@NickLaMuro sure, will do! Thank you for info!

**Comment by AlexanderZagaynov** [comment]
@Fryguy can this be reopened too?

**Comment by AlexanderZagaynov** [comment]
made a new one: https://github.com/ManageIQ/manageiq-gems-pending/pull/449

**Pull request closed without merging**